python/src/aiconfig/editor/client/src/components/EditorContainer.tsx

```diff
@@ -665,9 +669,45 @@ export default function EditorContainer({
     return () => window.removeEventListener("keydown", saveHandler);
   }, [onSave]);
 
+  // Server heartbeat, check every 3s to show error if server is down
+  const getServerStatusCallback = callbacks.getServerStatus;
+  useEffect(() => {
+    if (!getServerStatusCallback) {
+      return;
+    }
+
+    const interval = setInterval(async () => {
+      try {
+        const res = await getServerStatusCallback();
+        setServerStatus(res.status);
+      } catch (err: unknown) {
+        setServerStatus("ERROR");
+      }
+    }, 3000);
```

Comment:
Can we store this in the same place with the DEBOUNCE param? Will make it easier to see our macros

Reply:
Also is there a best practice for how frequent heartbeats should be? Or similar to debounce it's just more of a "play around and find out" kind of deal?

Reply:
I think it's usually order of seconds but depends on the importance. In our case there's a potential for lost work so shouldn't be a large interval